src/ppxgboost/PPPrediction.py

```diff
@@ -0,0 +1,129 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pandas as pd
+
+import ppxgboost.PaillierAPI as paillier
+from ppxgboost.PPModel import *
+from ppxgboost.PPQuery import PPQuery
+
+
+def predict_single_input_binary(model: PPModel, query, default_base_score=0.5):
+    """
+    Return the score for this query in a binary classification model
+
+    :param model: the model to evaluate
+    :param query: a single input query
+    :param default_base_score: a default score is 0.5 (global bias)
+    :return: the query's score
+    """
+
+    scores = model.eval(query)
+    return default_base_score + sum(scores)
+
+def predict_binary(enc_model: PPModel, enc_queries: [PPQuery], default_base_score=0.5):
+    """
+    Prediction on @enc_queries over the @enc_model
+    :param enc_model: encrypted PPModel
+    :param enc_queries: a list of encrypted PPQuery
+    :param default_base_score: default score is 0.5 (according to the xgboost -- global bias)
+    :return: a list of prediction values
+    """
+
+    return list(map(lambda q: predict_single_input_binary(enc_model, q, default_base_score), enc_queries))
```

Comment:
Not a big deal, but doesn't map() return a list already?

Reply:
No, `map` returns a weird data structure, which e.g., you can't call `len()` on.